cmd/minishift/cmd/stop.go

```diff
@@ -61,12 +61,16 @@ func runStop(cmd *cobra.Command, args []string) {
 
 	fmt.Println("Stopping local OpenShift cluster...")
 
-	if !skipUnRegistration {
-		if _, err := cluster.UnRegister(api); err != nil {
+	if !skipUnRegistration && minishiftConfig.InstanceConfig.IsRegistered {
+		supportUnregistration, err := cluster.UnRegister(api)
```

Comment:
Also, the whole point of 'minishiftConfig.InstanceConfig.IsRegistered' was to be able to handle the un-regristration case, for the case that the VM cannot be accessed. So assuming all data is in a consistent state the check for _minishiftConfig.InstanceConfig.IsRegistered_ already tells us that the VM supports (un)registration. So I think all that's needed is to handle the error a bit differently. 

We might want to align with the _delete_ command where in the case of a failing un-registration the user gets prompted on whether to processed or not. See https://github.com/minishift/minishift/blob/2335e6ac4aa252e7046070f2f8a5ca76675f12c3/cmd/minishift/cmd/delete.go#L54